esp/esp/program/modules/handlers/studentregphasezeromanage.py

```diff
@@ -71,23 +71,20 @@ def lottery(self, prog, role):
         # The lottery algorithm is run, with randomization and processing in order.
         # If any one in the group doesn't get in (due to cap size), no one in that group gets in.
         counts = {key:0 for key in grade_caps}
-        winners = Group.objects.get_or_create(name=role)[0]
+        winners, _ = Group.objects.get_or_create(name=role)
 
         for i in records:
             sibs = i.user.all()
-            newcounts = copy.copy(counts)
+            newcounts = counts
```

Comment:
Ah! So in that case you do want to do the copy at the beginning (but no need at the end). Sorry for leading you astray, I didn't read carefully enough.